src/libcore/char.rs

```diff
@@ -299,7 +299,17 @@ impl CharExt for char {
 
     #[inline]
     fn escape_unicode(self) -> EscapeUnicode {
-        EscapeUnicode { c: self, state: EscapeUnicodeState::Backslash }
+        let c = self as u32;
+        // or-ing 1 ensures that for c==0 the code computes that one
+        // digit should be printed and (which is the same) avoids the
+        // (31 - 32) underflow
+        let msb = 31 - (c | 1).leading_zeros();
+        let msdigit = msb / 4;
```

Comment:
This code is pretty confusing (I had to re-interpret it several times), so bear with me…

Should this variable be named `nibbles`/`hex_digits` (to-be-output) or something along the lines?